### DISCUSSION TITLE
NV-Control extension alternative for Wayland

### DISCUSSION THREAD

**Discussion by phodina** [NVIDIA/open-gpu-kernel-modules#262] [Q&A] [answered]
Hi,

I'd like to ask what interface to use when running on Wayland. 

Before I used [NV-Control](https://github.com/NVIDIA/nvidia-settings/blob/main/doc/NV-CONTROL-API.txt) extension on X11, however, this is not possible on Wayland enabled desktop.

Do you plan to support it in [nvidia-settings](https://github.com/NVIDIA/nvidia-settings/issues/69)?

**Comment by phodina** [comment]
The reason why I ask is that I'd like to get [Green with Envy](https://gitlab.com/leinardi/gwe/-/issues/154) working on Wayland.

**Comment by erik-kz** [comment, marked as answer]
You might want to look into the NVML library https://docs.nvidia.com/deploy/nvml-api/index.html

It interfaces directly with our kernel module, so it will work regardless of whether you're using X11 or Wayland. It's the basis for the nvidia-smi command line utility. Our long-term plan for Wayland support in nvidia-settings is to migrate as much functionality as possible from NV-CONTROL to NVML. It's not exactly a one-to-one replacement, but broadly speaking the feature sets are pretty similar.

**Reply by arisu3** [reply]
There are a lot of features in NV-CONTROL that are not supported on NVML and I see no signs that there are efforts to add support. This includes things like setting GPU clock offsets, which is necessary for underclocking on Linux.
